azurerm/internal/services/analysisservices/tests/analysis_services_server_resource_test.go

```diff
@@ -210,6 +211,33 @@ func TestAccAzureRMAnalysisServicesServer_backupBlobContainerUri(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMAnalysisServicesServer_suspended(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_analysis_services_server", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMAnalysisServicesServerDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAnalysisServicesServer_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAnalysisServicesServerExists(data.ResourceName),
+					testSuspendAzureRMAnalysisServicesServer(data.ResourceName),
```

Comment:
would it make sense to add a `state` property to the resource so we don't have to do this?

Reply:
Was thinking about that but didn't find a real use case for that attribute. Didn't want to add it simply to make a test easier.

Reply:
we need to confirm the state in Azure, rather than in the Statefile - so it's better to check Azure directly here